packages/consumption/src/modules/notifications/itemProcessors/AttributeDeletedNotificationItemProcessor.ts

```diff
@@ -0,0 +1,73 @@
+import { ILogger } from "@js-soft/logging-abstractions";
+import { AttributeDeletedNotificationItem } from "@nmshd/content";
+import { CoreDate, CoreErrors as TransportCoreErrors, TransportLoggerFactory } from "@nmshd/transport";
+import { ConsumptionController } from "../../../consumption/ConsumptionController";
+import { CoreErrors } from "../../../consumption/CoreErrors";
+import { AttributeDeletedByPeerEvent, LocalAttribute } from "../../attributes";
+import { LocalAttributeDeletionStatus } from "../../attributes/local/LocalAttributeDeletionStatus";
+import { ValidationResult } from "../../common";
+import { LocalNotification } from "../local/LocalNotification";
+import { AbstractNotificationItemProcessor } from "./AbstractNotificationItemProcessor";
+
+export class AttributeDeletedNotificationItemProcessor extends AbstractNotificationItemProcessor<AttributeDeletedNotificationItem> {
+    private readonly _logger: ILogger;
+
+    public constructor(consumptionController: ConsumptionController) {
+        super(consumptionController);
+        this._logger = TransportLoggerFactory.getLogger(AttributeDeletedNotificationItemProcessor);
+    }
+
+    public override async checkPrerequisitesOfIncomingNotificationItem(
+        notificationItem: AttributeDeletedNotificationItem,
+        notification: LocalNotification
+    ): Promise<ValidationResult> {
+        const attribute = await this.consumptionController.attributes.getLocalAttribute(notificationItem.attributeId);
+
+        if (typeof attribute === "undefined") {
+            return ValidationResult.error(TransportCoreErrors.general.recordNotFound(LocalAttribute, notificationItem.attributeId.toString()));
+        }
+
+        if (!attribute.isOwnSharedAttribute(this.currentIdentityAddress)) {
+            return ValidationResult.error(CoreErrors.attributes.isNotOwnSharedAttribute(notificationItem.attributeId));
+        }
+
+        if (!notification.peer.equals(attribute.shareInfo.peer)) {
+            return ValidationResult.error(CoreErrors.attributes.senderIsNotPeerOfSharedAttribute(notification.peer, notificationItem.attributeId));
+        }
+
+        return ValidationResult.success();
+    }
+
+    // TODO: add deletion of predecessors
+    public override async process(notificationItem: AttributeDeletedNotificationItem, notification: LocalNotification): Promise<AttributeDeletedByPeerEvent> {
```

Comment:
Here and below in the rollback function I get an eslint error, that the parameter `notification` is not used. It is required by the parent class, but I don't need it. Should we make in optional in the parent class?

Reply:
You have two options:
1. just remove this parameter **here** (please not make it optional in the parent..), overriding functions can have less params in ts, so this compiles perfectly
2. mark the parameter with an underscore (`_notification: LocalNotification`) to show eslint that you know it's unused